solidity/test/bridge/Bridge.test.ts

```diff
@@ -2233,229 +2252,1840 @@ describe("Bridge", () => {
+                          await expect(tx)
+                            .to.emit(bank, "BalanceDecreased")
+                            .withArgs(bridge.address, 0)
+                          // ...nor changed in any other way.
+                          expect(
+                            bridgeBalance.afterProof.sub(
+                              bridgeBalance.beforeProof
+                            )
+                          ).to.equal(0)
+                        })
+
+                        it("should not transfer anything to the treasury", async () => {
+                          // Treasury balance should not be increased in any way
+                          // since the only request handled by the redemption
+                          // transaction is reported as timed out and is just
+                          // skipped during processing.
+                          expect(
+                            treasuryBalance.afterProof.sub(
+                              treasuryBalance.beforeProof
+                            )
+                          ).to.equal(0)
+                        })
+
+                        it("should not change redeemer balance in any way", async () => {
+                          // Redemption proof submission should NEVER cause
+                          // any change of the redeemer balance.
+                          const redeemerBalance = redeemersBalances[0]
+
+                          expect(
+                            redeemerBalance.afterProof.sub(
+                              redeemerBalance.beforeProof
+                            )
+                          ).to.be.equal(0, "Balance of redeemer has changed")
+                        })
+                      }
+                    )
+
+                    context(
+                      "when the single output is a pending requested redemption but amount is wrong",
+                      () => {
+                        const data: RedemptionTestData = JSON.parse(
+                          JSON.stringify(SinglePendingRequestedRedemption)
+                        )
+
+                        let outcome: Promise<RedemptionScenarioOutcome>
+
+                        before(async () => {
+                          await createSnapshot()
+
+                          // Alter the single redemption request in the test
+                          // data and set such an amount that will cause
+                          // the Bitcoin redemption transaction to be deemed
+                          // as invalid due to a wrong amount. The transaction
+                          // output has the value of 1176924 so to make this
+                          // test scenario happen, the request amount must be
+                          // way different (lesser or greater) than the output
+                          // value.
+                          data.redemptionRequests[0].amount = 300000
```

Comment:
Yeah but I think we have quite good documentation in the contract explaining how the correct redemption range is defined. I had to use some simplification in this test scenario because it's not so straightforward to test both edge cases at once since the requested amount is also validated against the wallet balance which is determined using an actual testnet BTC UTXO. That means we need to assemble two separate and quite tricky BTC transactions and test data for it. This is why I just tested this case by reusing `SinglePendingRequestedRedemption` data set and just checked the amount condition in a general way.

Reply:
I see... Can you add a simplified version of this explanation to the comment? Makes sense to me and I think it's a good-to-know for future-us.

Reply:
See https://github.com/keep-network/tbtc-v2/pull/155/commits/66841439336e820b3e582905bb08be09458e02e5